### PR TITLE
Remove parcelable objects

### PR DESCRIPTION
**Pull request by amit-metri**
-updated covid test details data fetching logic
-bug fix on scrolling of covid test details

### CODE REVIEW

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/covidtests/SingleTestResultFragment.kt [hunk 1]
```diff
@@ -246,11 +282,11 @@ class SingleTestResultFragment : Fragment(R.layout.fragment_single_test_result)
     companion object {
 
         @JvmStatic
-        fun newInstance(testRecordDto: TestRecordDto, patientDto: PatientDto) =
+        fun newInstance(testRecordId: String, testResultId: Long) =
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/covidtests/SingleTestResultFragment.kt:285]: testRecordId must be long

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: data/src/main/java/ca/bc/gov/data/datasource/PatientLocalDataSource.kt [hunk 2]
```diff
@@ -120,4 +124,8 @@ class PatientLocalDataSource @Inject constructor(
 
     suspend fun isAuthenticatedPatient(patientId: Long): Boolean =
         patientDao.isAuthenticatedPatient(patientId) > 0
+
+    suspend fun getPatientDto(patientId: Long): PatientDto {
```
[the remark this answers is not in the document]
Comment (amit-metri) [reply data/src/main/java/ca/bc/gov/data/datasource/PatientLocalDataSource.kt]: using getPatientWithTestResultAndRecords to get the data

File: repository/src/main/java/ca/bc/gov/repository/patient/PatientRepository.kt [hunk 3]
```diff
@@ -98,4 +105,7 @@ class PatientRepository @Inject constructor(
 
     suspend fun isAuthenticatedPatient(patientId: Long): Boolean =
         patientLocalDataSource.isAuthenticatedPatient(patientId)
+
+    suspend fun getPatientDto(patientId: Long): PatientDto =
```
[the remark this answers is not in the document]
Comment (amit-metri) [reply repository/src/main/java/ca/bc/gov/repository/patient/PatientRepository.kt]: using getPatientWithTestResultAndRecords to get the data

File: data/src/main/java/ca/bc/gov/data/local/dao/PatientDao.kt [hunk 4]
```diff
@@ -56,6 +57,10 @@ interface PatientDao {
     @Query("SELECT * FROM test_result WHERE id = :testResultId")
     suspend fun getPatientWithTestResultAndRecords(testResultId: Long): TestResultWithRecordsAndPatient?
 
+    @Transaction
+    @Query("SELECT * FROM test_record WHERE id = :testRecordId")
+    suspend fun getTestRecord(testRecordId: String): TestRecordEntity?
```
[the remark this answers is not in the document]
Comment (amit-metri) [reply data/src/main/java/ca/bc/gov/data/local/dao/PatientDao.kt]: using getPatientWithTestResultAndRecords to get the data

File: app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/covidtests/SingleTestResultFragment.kt (hunk 1 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/covidtests/SingleTestResultFragment.kt:285]: using getPatientWithTestResultAndRecords to get the data

**Review by pinakinkansara — APPROVED** [review]

**Pull request merged**